### PR TITLE
Model Accept-Patch header

### PR DESCRIPTION
**Pull request by cquiroz**
As part of modeling headers, here's `Accept-Patch`

### CODE REVIEW

File: core/src/main/scala/org/http4s/Header.scala [hunk 1]
```diff
@@ -120,6 +120,17 @@ object Header {
     }
   }
 
+  /** Helper trait that provides a default way of rendering the value provided a Renderer */
+  trait RecurringRenderer extends Recurring {
```
Comment (cquiroz) [inline core/src/main/scala/org/http4s/Header.scala:124]: This is equivalent to `RecurringRenderable` but for `Renderer

File: core/src/main/scala/org/http4s/Header.scala [hunk 2]
```diff
@@ -120,6 +120,17 @@ object Header {
...
+    type Value
+    implicit val renderer: Renderer[Value]
```
Comment (cquiroz) [inline core/src/main/scala/org/http4s/Header.scala:126]: Any ideas on how to initialize this by default? My attempts ended poorly 😄

File: tests/src/test/scala/org/http4s/HeadersSpec.scala [hunk 3]
```diff
@@ -46,7 +46,7 @@ class HeadersSpec extends Http4sSpec {
         Header("Accept-Patch", ""),
         Header("Access-Control-Allow-Credentials", "")
       )
-      headers.get(`Accept-Patch`).map(_.value) must beSome("")
+      headers.get(`Access-Control-Allow-Credentials`).map(_.value) must beSome("")
```
Comment (cquiroz) [inline tests/src/test/scala/org/http4s/HeadersSpec.scala:49]: I'm not sure about the meaning of this test

File: core/src/main/scala/org/http4s/Header.scala (hunk 2 above)
Comment (rossabaker) [reply core/src/main/scala/org/http4s/Header.scala:126]: Eager abstract values make me nervous. How about a def?

File: tests/src/test/scala/org/http4s/HeadersSpec.scala (hunk 3 above)
Comment (rossabaker) [reply tests/src/test/scala/org/http4s/HeadersSpec.scala:49]: I think it was just making sure that we could get headers when we had a key but not a full model.  I don't think it's a great test, and what you did is fine.

File: core/src/main/scala/org/http4s/Header.scala (hunk 2 above)
Comment (cquiroz) [reply core/src/main/scala/org/http4s/Header.scala:126]: sure

**Comment by cquiroz** [comment]
Rebased and fixed comment

**Pull request merged**